runtime/compiler/env/j9method.cpp

```diff
@@ -6757,6 +6757,39 @@ TR_ResolvedJ9Method::getResolvedDynamicMethod(TR::Compilation * comp, I_32 callS
 #else
    TR_ResolvedMethod *result = NULL;
 
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   TR_ResolvedMethod * owningMethod = comp->getCurrentMethod();
+   J9InvokeCacheEntry *invokeCache = (J9InvokeCacheEntry *) owningMethod->callSiteTableEntryAddress(callSiteIndex);
+   if (!invokeCache)
+      comp->failCompilation<TR::CompilationException>("unable to retrieve side table entry for invokeDynamic");
+      {
+      TR::VMAccessCriticalSection getResolvedDynamicMethod(fej9());
+
+      J9Class    *ramClass = constantPoolHdr();
+      J9ROMClass *romClass = romClassPtr();
+
+      if (ramClass->callSites[callSiteIndex] == NULL)
+         {
+         J9SRP                 *namesAndSigs = (J9SRP*)J9ROMCLASS_CALLSITEDATA(romClass);
+         J9ROMNameAndSignature *nameAndSig   = NNSRP_GET(namesAndSigs[callSiteIndex], J9ROMNameAndSignature*);
+         J9UTF8                *signature    = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
+
+         TR_OpaqueMethodBlock *dummyInvoke = _fe->getMethodFromName("java/lang/invoke/MethodHandle", "linkToStatic", "([Ljava/lang/Object;)Ljava/lang/Object;");
```

Comment:
I think the dummy invoke is the only thing different from current implementation, other code can be shared